debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9ConstantPoolCommand.java

```diff
@@ -0,0 +1,539 @@
+						J9RAMFieldRefPointer ref = J9RAMFieldRefPointer.cast(ramCPP);
+						if ((staticRef.flagsAndClass().longValue() > 0) && !staticRef.valueOffset().eq(UDATA.MAX)) {
+							ramInfo = staticRef.valueOffset().getHexValue() + " (Static offset)"; // resolved static
+						} else if (ref.flags().gt(ref.valueOffset())) {
+							ramInfo = ref.valueOffset().getHexValue() + " (Instance offset)"; // resolved instance
+						} else {
+							ramInfo = "(unresolved)"; // unresolved
+						}
+					}
+
+					J9ROMFieldRefPointer romFieldRef = J9ROMFieldRefPointer.cast(romCPP);
+					romInfo = "#" + Long.toString(romFieldRef.classRefCPIndex().longValue());
+
+					J9ROMClassRefPointer classRef = J9ROMClassRefPointer.cast(romConstantPool.add(romFieldRef.classRefCPIndex()));
+					J9ROMNameAndSignaturePointer nameAndSig = romFieldRef.nameAndSignature();
+					value = String.format("%s.%s:%s #%d",
+											J9UTF8Helper.stringValue(classRef.name()),
+											J9UTF8Helper.stringValue(nameAndSig.name()),
+											J9UTF8Helper.stringValue(nameAndSig.signature()),
+											romFieldRef.classRefCPIndex().intValue());
+				} else if ((cpType == J9ConstantPool.J9CPTYPE_STATIC_METHOD) ||
+						   (cpType == J9ConstantPool.J9CPTYPE_INTERFACE_STATIC_METHOD) ||
+						   (cpType == J9ConstantPool.J9CPTYPE_INTERFACE_INSTANCE_METHOD) ||
+						   (cpType == J9ConstantPool.J9CPTYPE_INSTANCE_METHOD)) {
+
+					J9ROMMethodRefPointer romMethodRef = J9ROMMethodRefPointer.cast(romCPP);
+					J9ROMClassRefPointer romClassRef = J9ROMClassRefPointer.cast(romConstantPool.add(romMethodRef.classRefCPIndex()));
+					J9ROMNameAndSignaturePointer nameAndSignature = romMethodRef.nameAndSignature();
+					String className = J9UTF8Helper.stringValue(romClassRef.name());
+					boolean inRamAndResolved = false;
+
+					if (i < ramCPCount) {
+						J9RAMMethodRefPointer ramMethodRef = J9RAMMethodRefPointer.cast(ramCPP);
+						J9MethodPointer ramMethod = J9RAMMethodRefPointer.cast(ramCPP).method();
+						long methodIndex = ramMethodRef.methodIndexAndArgCount().longValue() >> 8;
+
+						boolean isInitialMethod = true;
+						if ((cpType == J9ConstantPool.J9CPTYPE_STATIC_METHOD) ||
+							(cpType == J9ConstantPool.J9CPTYPE_INTERFACE_STATIC_METHOD)) {
+							if ((ramMethod.getAddress() != initializerMethods.initialStaticMethod().getAddress()) ||
+								(methodIndex != J9VTABLE_INITIAL_VIRTUAL_OFFSET)) {
+								isInitialMethod = false;
+							}
+						} else {
+							if (ramMethod.getAddress() != initializerMethods.initialSpecialMethod().getAddress()) {
+								isInitialMethod = false;
+							} else {
+								if (methodIndex != J9VTABLE_INITIAL_VIRTUAL_OFFSET) { // virtual method
+									J9ClassPointer receiverClass = J9RAMClassRefPointer.cast(ramConstantPool.add(romMethodRef.classRefCPIndex())).value();
+									if (receiverClass.notNull()) {
+										long methodAddress = PointerPointer.cast(UDATA.cast(receiverClass).add(methodIndex)).at(0).longValue();
```

Comment:
Use `getAddress()` instead of `longValue()` to avoid problems with pointers where the sign bit is set.